rpcs3/Emu/RSX/Common/BufferUtils.cpp

```diff
@@ -8,15 +8,57 @@
 
 #define DEBUG_VERTEX_STREAMING 0
 
-const bool s_use_ssse3 =
-#ifdef _MSC_VER
-	utils::has_ssse3();
-#elif __SSSE3__
-	true;
+// Clang unconditionally includes the header files defining the intrinsics even when the specified arch does not support them, then refuses to compile
+// Forces use of ugly workarounds with macro redirection
```

Comment:
That's not really necessary to redefine macro. All we need is a function that does what the intrinsic does. Its body can conditionally fallback to assembly if an instruction set is not available.

Reply:
GCC allowed to actually override the intrinsics hence my frustration with clang there, which made it so that with GCC I could actually define _mm_shuffle_epi8 itself making it a drop-in replacement. Clang did not like that.